chunk/dynamo_table_manager.go

```diff
@@ -0,0 +1,347 @@
+package chunk
+
+import (
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"github.com/weaveworks/common/mtime"
+	"github.com/weaveworks/scope/common/instrument"
+	"golang.org/x/net/context"
+)
+
+const (
+	readLabel        = "read"
+	writeLabel       = "write"
+	minWriteCapacity = 1
+)
+
+var (
+	syncTableDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "cortex",
+		Name:      "dynamo_sync_tables_seconds",
+		Help:      "Time spent doing syncTables.",
+		Buckets:   prometheus.DefBuckets,
+	}, []string{"operation", "status_code"})
+	tableCapacity = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "cortex",
+		Name:      "dynamo_table_capacity_units",
+		Help:      "Per-table DynamoDB capacity, measured in DynamoDB capacity units.",
+	}, []string{"op", "table"})
+)
+
+func init() {
+	prometheus.MustRegister(tableCapacity)
+}
+
+// TableManagerConfig is the config for a DynamoTableManager
+type TableManagerConfig struct {
+	DynamoDBURL          string
+	DynamoDBPollInterval time.Duration
+
+	UsePeriodicTables    bool
+	TablePrefix          string
+	TablePeriod          time.Duration
+	PeriodicTableStartAt time.Time
+
+	// duration a table will be created before it is needed.
+	CreationGracePeriod        time.Duration
+	MaxChunkAge                time.Duration
+	ProvisionedWriteThroughput int64
+	ProvisionedReadThroughput  int64
+
+	// Not exported as only used by tests to inject mocks
+	dynamodb dynamodbClient
```

Comment:
The `MakeDynamoDbClient` function can still be defined here. That'd keep main.go fairly small.